src/test/kotlin/uk/gov/justice/digital/hmpps/deliusapi/service/person/DisabilityServiceTest.kt

```diff
@@ -0,0 +1,164 @@
+
+  @InjectMocks
+  lateinit var subject: DisabilityService
+
+  private lateinit var offender: Offender
+  private lateinit var request: NewDisability
+  private lateinit var disability: Disability
+
+  private var crn = Fake.crn()
+
+  @BeforeEach
+  fun beforeEach() {
+    request = Fake.newDisability()
+    crn = Fake.crn()
+    disability = Fake.disability()
+  }
+
+  @Test
+  fun `attempting to insert disability with missing offender`() {
+    havingOffenderByCrn(false)
+
+    val exception = Assertions.assertThrows(NotFoundException::class.java) {
+      whenCreatingDisability()
+    }
+    Assertions.assertEquals("Offender with crn '$crn' cannot be found", exception.message)
+  }
+
+  @Test
+  fun `attempting to insert disability with finish date before start date`() {
+    havingOffenderByCrn(true)
+    havingDisabilityTypeCode(request.disabilityTypeCode, havingReferenceCode = true, selectable = true)
+    havingDisabilityFinishDateBeforeStartDate()
+
+    val exception = Assertions.assertThrows(BadRequestException::class.java) {
+      whenCreatingDisability()
+    }
+    Assertions.assertEquals("The end date cannot be earlier than the start date.", exception.message)
+  }
+
+  @Test
+  fun `attempting to insert disability with invalid disability type`() {
+    havingOffenderByCrn(true)
+    havingDisabilityTypeCode(request.disabilityTypeCode, havingReferenceCode = false, selectable = true)
+    havingDisabilityFinishDateBeforeStartDate()
+
+    val exception = Assertions.assertThrows(BadRequestException::class.java) {
+      whenCreatingDisability()
+    }
+    Assertions.assertEquals("An invalid Disability Type was provided.", exception.message)
+  }
+
+  @Test
+  fun `attempting to insert disability of the same type as existing active disability`() {
+    havingOffenderByCrn(true)
+    havingDisabilityTypeCode(request.disabilityTypeCode, havingReferenceCode = true, selectable = true)
+    havingActiveDisabilityOfTheSameType(true)
+
+    val exception = Assertions.assertThrows(BadRequestException::class.java) {
+      whenCreatingDisability()
+    }
+    Assertions.assertEquals("An active disability record of this type already exists - unable to insert new disability.", exception.message)
+  }
+
+  @Test
+  fun `disability created when called with valid data`() {
+    havingOffenderByCrn(true)
+    havingDisabilityTypeCode(request.disabilityTypeCode, havingReferenceCode = true, selectable = true)
+    havingActiveDisabilityOfTheSameType(false)
+    havingSavedDisability()
+
+    whenCreatingDisability()
+    shouldSaveDisability()
+  }
+
+  private fun havingOffenderByCrn(offenderExists: Boolean = true) {
+    offender = Fake.offender().also {
+      it.crn = crn
+    }
+    whenever(offenderRepository.findByCrn(crn))
+      .thenReturn(if (offenderExists) offender else null)
+  }
+
+  private fun havingDisabilityTypeCode(
+    code: String,
+    havingReferenceCode: Boolean,
+    selectable: Boolean
```

Comment:
There's quite a bit of duplication above of `havingReferenceCode = true, selectable = true`.  Would the code read better if you added some defaults here?
```suggestion
    code: String = request.disabilityTypeCode,
    havingReferenceCode: Boolean = true,
    selectable: Boolean = true
```